meson.build

```diff
@@ -0,0 +1,141 @@
+global_configuration.set_quoted('SYSCONFDIR', get_option('sysconfdir'))
+
+config_header = configure_file(
+	output: 'config.h',
+	configuration: global_configuration,
+)
+
+config_dep = declare_dependency(
+	sources: config_header,
+	include_directories: include_directories('.')
+)
+
+if get_option('python-backend')
+	python = import('python')
+	python_exec = python.find_installation()
+	python_package_dir = get_option('python-package-dir')
+	if python_package_dir == ''
+		python_package_dir = python_exec.get_install_dir()
+	endif
+	python_package_dir = join_paths(python_package_dir, 'packagekit')
+else
+	python_package_dir = 'disabled'
+endif
+
+pk_db_dir = join_paths(get_option('localstatedir'), 'lib', 'PackageKit')
+local_state_dir = get_option('localstatedir')
+test_data_dir = join_paths(meson.source_root(), 'data', 'tests')
+package_data_dir = get_option('datadir')
+package_locale_dir = join_paths(get_option('prefix'), get_option('datadir'), 'locale')
+pk_plugin_dir = join_paths(get_option('prefix'), get_option('libdir'), 'packagekit-backend')
+
+# work both in C and C++
+add_project_arguments(
+	'-Wall',
+	'-Wcast-align', '-Wno-uninitialized',
+	'-Wmissing-declarations',
+#	'-Wredundant-decls',
+	'-Wpointer-arith',
+	'-Wcast-align',
+	'-Wwrite-strings',
+	'-Winit-self',
+	'-Wreturn-type',
+	'-Wformat-nonliteral',
+	'-Wformat-security',
+	'-Wmissing-include-dirs',
+	'-Wmissing-format-attribute',
+#	'-Wclobbered',
+	'-Wempty-body',
+#	'-Wignored-qualifiers',
+	'-Wsign-compare',
+#	'-Wtype-limits',
+	'-Wuninitialized',
+	language: ['c', 'cpp']
+)
+
+# work only in C
+add_project_arguments(
+	'-Waggregate-return',
+	'-Wdeclaration-after-statement',
+	'-Wshadow',
+	'-Wno-strict-aliasing',
+	'-Winline',
+#	'-Wmissing-parameter-type',
+#	'-Woverride-init',
+	language: 'c'
+)
+
+subdir('po')
+subdir('policy')
+subdir('etc')
+subdir('data')
+subdir('lib')
+subdir('src')
+subdir('client')
+subdir('backends')
+
+message('''
+	=== Project Configuration ===
+prefix:                    @0@
+datadir:                   @1@
+Able to run locally:       @2@
+Use elogind:               @3@
+Use systemd:               @4@
+Enable offline update:     @5@
+GStreamer plugin:          @6@
+Pango module (GTK+3):      @7@
+gobject-introspection:     @8@
+BASH Command not found:    @9@
+Cron scripts:              @10@
+python3:                   @11@
+Installing python backend: @12@
+python package dir:        @13@
+Backend:                   @14@
+	============================='''.format(
+	get_option('prefix'),
+	get_option('datadir'),
+	get_option('local-checkout'),
+	get_option('elogind'),
+	get_option('systemd'),
+	get_option('offline-update'),
+	get_option('gstreamer-plugin'),
+	get_option('gtk-module'),
+	get_option('gobject-introspection'),
+	get_option('bash-command-not-found'),
+	get_option('cron'),
+	get_option('python-3'),
+	get_option('backend-name'),
+	get_option('python-backend'),
```

Comment:
`backend-name` and `python-backend` need to be swapped here. Also, there's nothing that will make it print the selected DNF vendor if the DNF backend is chosen?

Reply:
@Conan-Kudo Made it adaptive to the backend and added dnf vendor message in case the dnf backend is enabled